OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -333,27 +333,23 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 							break;
 
 						if (order.GroupedActors == null)
-							ResolveOrder(order, world.WorldActor.TraitsImplementing<IValidateOrder>(), orderManager, clientId);
+							ResolveOrder(order, world, world.OrderValidators, orderManager, clientId);
 						else
-						{
-							// PERF: Cache the result of TraitsImplementing as we are likely to use it for several order subjects
-							var validateOrders = world.WorldActor.TraitsImplementing<IValidateOrder>().ToArray();
 							foreach (var subject in order.GroupedActors)
-								ResolveOrder(Order.FromGroupedOrder(order, subject), validateOrders, orderManager, clientId);
-						}
+								ResolveOrder(Order.FromGroupedOrder(order, subject), world, world.OrderValidators, orderManager, clientId);
 
 						break;
 					}
 			}
 		}
 
-		static void ResolveOrder(Order order, IEnumerable<IValidateOrder> validateOrders, OrderManager orderManager, int clientId)
+		static void ResolveOrder(Order order, World world, IEnumerable<IValidateOrder> validateOrders, OrderManager orderManager, int clientId)
```

Comment:
You can remove the `IEnumerable<IValidateOrder> validateOrders` parameter now, since we can get that from `world`.